Calculator/Calculator/Controller/ViewController.swift

```diff
@@ -7,12 +7,126 @@
 import UIKit
 
 class ViewController: UIViewController {
-
+    @IBOutlet weak var operatorLabel: UILabel!
+    @IBOutlet weak var operandLabel: UILabel!
+    @IBOutlet weak var stackView: UIStackView!
+    @IBOutlet weak var scrollView: UIScrollView!
+    private var expression: String = ""
+    private var isResult: Bool = false
+    private var isInputZero: Bool = true
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        clearLabel()
     }
 
+    @IBAction func touchUpAllClearButton(_ sender: UIButton) {
+        stackView.subviews.forEach { $0.removeFromSuperview() }
+        clearLabel()
+        expression = ""
+        configureCondition(isResult: false, isInputZero: true)
+    }
+    
+    @IBAction func touchUpCleanEntryButton(_ sender: UIButton) {
+        guard !isResult else {
+            return
+        }
+        
+        operandLabel.text = "0"
+        configureCondition(isResult: false, isInputZero: false)
+    }
+    
+    @IBAction func touchUpSignButton(_ sender: UIButton) {
+        guard let operand = operandLabel.text, operand != "0", operand != "0." else {
+            return
+        }
+        guard operand.hasPrefix("-") else {
+            operandLabel.text = "-\(operand)"
+            return
+        }
+        
+        operandLabel.text = String(operand.suffix(operand.count - 1))
+    }
+    
+    @IBAction func touchUpResultButton(_ sender: UIButton) {
+        guard !isResult else {
+            return
+        }
+        
+        addStackView()
+        var formula: Formula = ExpressionParser.parse(from: expression)
+        let result: Double = formula.result()
+        operandLabel.text = result.formatNumbers()
+        operatorLabel.text = ""
+        expression = ""
+        configureCondition(isResult: true, isInputZero: true)
+    }
+    
+    @IBAction func touchUpOperandButton(_ sender: UIButton) {
+        guard let currentOperand = sender.currentTitle, !isResult else {
+            return
+        }
+        guard currentOperand != "0" else {
+            configureCondition(isResult: false, isInputZero: true)
+            return
+        }
+        guard let operand = operandLabel.text, !(operand.hasSuffix(".") && currentOperand == ".") else {
+            return
+        }
+        guard operand != "0" || currentOperand == "." else {
+            operandLabel.text = currentOperand
+            return
+        }
+
+        operandLabel.text = operand + currentOperand
+    }
+    
+    @IBAction func touchUpOperatorButton(_ sender: UIButton) {
+        guard let operand = operandLabel.text, operand != "0" || isInputZero else {
+            operatorLabel.text = sender.currentTitle
+            return
+        }
+        
+        if operand.hasSuffix(".") {
+            operandLabel.text = String(operand.prefix(operand.count - 1))
+        }
```

Comment:
[prefix(_:)](https://developer.apple.com/documentation/swift/string/prefix(_:)) 문서를 보면 시간 복잡도에 대한 이야기가 나와있습니다. 다음과 같이 처리하면 더 효율적일 것 같습니다.

```suggestion
        if operand.hasSuffix(".") {
            operandLabel.text?.removeLast()
        }
```

Reply:
`removeLast` 메서드를 사용하여 수정하였습니다.